From u.kleine-koenig@pengutronix.de  Mon Jun 21 14:39:28 2010
From: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
Date: Mon, 21 Jun 2010 16:11:45 +0200
Subject: Driver core: move platform device creation helpers to .init.text (if MODULE=n)
To: Greg KH <greg@kroah.com>
Cc: Greg Kroah-Hartman <gregkh@suse.de>, Magnus Damm <damm@opensource.se>, "Rafael J. Wysocki" <rjw@sisk.pl>, Uwe Kleine-König <u.kleine-koenig@pengutronix.de>, Paul Mundt <lethal@linux-sh.org>, linux-kernel@vger.kernel.org
Message-ID: <1277129511-2732-2-git-send-email-u.kleine-koenig@pengutronix.de>


Platform devices should only be called by init code, so it should be
possible to move creation helpers to .init.text -- at least if modules
are disabled.

Signed-off-by: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/platform.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/base/platform.c
+++ b/drivers/base/platform.c
@@ -357,7 +357,7 @@ EXPORT_SYMBOL_GPL(platform_device_unregi
  *
  * Returns &struct platform_device pointer on success, or ERR_PTR() on error.
  */
-struct platform_device *platform_device_register_resndata(
+struct platform_device *__init_or_module platform_device_register_resndata(
 		struct device *parent,
 		const char *name, int id,
 		const struct resource *res, unsigned int num,
